examples/apps/tic-tac-toe/src/Board.flix

```diff
@@ -0,0 +1,110 @@
+///
+/// Struct representing the board, backed by an immutable vector.
+///
+
+pub struct Board[r] {
+    vec: Vector[Symbol]
+}
+
+
+mod Board {
+
+    ///
+    /// Returns a new, empty tic-tac-toe board.
+    ///
+    pub def newBoard(rc: Region[r]): Board[r] \ r = {
+        new Board @ rc {vec = Vector.repeat(9, Symbol.Empty)}
+    }
+
+    ///
+    /// Returns the current state of the board as a vector.
+    ///
+    pub def getVec(board: Board[r]): Vector[Symbol] = {
+        board->vec
+    }
+
+    ///
+    /// Makes a move on the board at the specified index with the given symbol.
+    ///
+    /// Returns a tuple containing the updated board and a boolean indicating whether the move was successful.
+    ///
+    pub def makeMove(board: Board[r], symbol: Symbol, index: Int32, rc: Region[r]): (Board[r], Bool) \ r = {
```

Comment:
Returning a Boolean for success is a bit unusual for effect-oriented programming, I suspect.

What happens if we instead introduce an error for this case?